### PR TITLE
in GSQ component throw user error on invalid endpoint url

### PR DESCRIPTION
**Pull request by imatiach-msft**
in GSQ component throw user error on invalid endpoint url

also adds some unit tests for GSQ component

### CODE REVIEW

**Review by ycheng35xo — APPROVED** [review]

File: assets/model_monitoring/components/tests/unit/test_gsq_histogram.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        url_pattern = AZURE_OPENAI_API_DEPLOYMENT_URL_PATTERN
+        domain_pattern_re = AZURE_ENDPOINT_DOMAIN_VALID_PATTERN_RE
+        version = "2022-12-01"
+        model = "test_model"
+        valid_url = "abc.openai.azure.com"
+        formatted_url = _check_and_format_azure_endpoint_url(
+            url_pattern, domain_pattern_re, valid_url, version, model)
+        expected_format = f"https://{valid_url}/openai/deployments/{model}?api-version={version}"
+        print(expected_format)
+        print(formatted_url)
```
Comment (gaugup) [inline assets/model_monitoring/components/tests/unit/test_gsq_histogram.py:46]: Do we want prints here?

**Review by gaugup — APPROVED** [review]

File: assets/model_monitoring/components/tests/unit/test_gsq_histogram.py (hunk 1 above)
Comment (imatiach-msft) [reply assets/model_monitoring/components/tests/unit/test_gsq_histogram.py]: good catch, removed!

File: assets/model_monitoring/components/src/generation_safety_quality/annotation_compute_histogram/run.py [hunk 2]
```diff
@@ -167,7 +168,9 @@ def _check_and_format_azure_endpoint_url(
         domain = domain[:-1]
 
     if not re.match(domain_pattern_re, domain):
-        raise RuntimeError(f"Invalid Azure endpoint domain URL: {domain}.")
+        err_msg = f"Invalid Azure endpoint domain URL: {domain}."
```
Comment (yunjie-hub) [inline assets/model_monitoring/components/src/generation_safety_quality/annotation_compute_histogram/run.py:171]: This is for the code change, are you going to manage the component release yourself?

**Review by yunjie-hub — APPROVED** [review]

File: assets/model_monitoring/components/src/generation_safety_quality/annotation_compute_histogram/run.py (hunk 2 above)
Comment (imatiach-msft) [reply assets/model_monitoring/components/src/generation_safety_quality/annotation_compute_histogram/run.py:171]: no not sure how to release it - I think I wanted to add a couple more tests/bug fixes before that as well

**Pull request merged**